celements-user-manager/component/src/main/java/com/celements/auth/user/validation/XWikiUsersValidationRule.java

```diff
@@ -0,0 +1,117 @@
+package com.celements.auth.user.validation;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import javax.inject.Inject;
+import javax.validation.constraints.NotNull;
+
+import org.springframework.stereotype.Component;
+import org.xwiki.model.reference.ClassReference;
+
+import com.celements.auth.user.User;
+import com.celements.auth.user.UserService;
+import com.celements.docform.DocFormRequestKey;
+import com.celements.docform.DocFormRequestParam;
+import com.celements.mailsender.IMailSenderRole;
+import com.celements.model.reference.RefBuilder;
+import com.celements.rights.access.EAccessLevel;
+import com.celements.rights.access.IRightsAccessFacadeRole;
+import com.celements.validation.IRequestValidationRule;
+import com.celements.validation.ValidationResult;
+import com.celements.validation.ValidationType;
+
+@Component
+public class XWikiUsersValidationRule implements IRequestValidationRule {
+
+  private IMailSenderRole mailSenderService;
+  private UserService userService;
+  private IRightsAccessFacadeRole rightsAccess;
```

Comment:
```suggestion
  private final IMailSenderRole mailSenderService;
  private final UserService userService;
  private final IRightsAccessFacadeRole rightsAccess;
```
injected beans should generally be defined as `final` for clarity.